### ISSUE TITLE
docs-bug(mat-tab-nav-panel): Documentation implies that [tabPanel] is optional, but must be specified.

### ISSUE THREAD

**Issue by yutamago** [angular/components#26270]
### Documentation Feedback

The documentation implies that [tabPanel] can be left unprovided by saying "If not provided, then the nav bar follows the ARIA link / navigation landmark pattern".

However, if left unprovided, it will throw an error stating: "A mat-tab-nav-panel must be specified via [tabPanel]".

### Affected documentation page

https://material.angular.io/components/tabs/api#MatTabNav

**Comment by epelc** [comment]
Please fix this. We have been using a standalone tab panel for navigation ie footer of our app for years. We use in combination with router links to go between pages when you click or activate a tab.

**Comment by epelc** [comment]
@wagnermaciel would you accept a PR for changes described in my comment?

**Comment by muenchto** [comment]
I agree with @epelc, these line should be removed to fit the documentation. 
Also, more complicated Apps can easily create a situation where the mat-tab-nav-panel is wrapped by some header component. In this case, providing the tabPanel via an Input does not work because of Angular's CD resolution and throws an ExpressionChangedAfterItHasBeenCheckedError. Bug reported in https://github.com/angular/components/issues/28379

**Comment by EliezerB123** [comment]
Does anyone know of any workaround to force [tabPanel] to be optional? 

We don't want to use [tabPanel] style tabs, because it'll cause us a lot of accessibility problems. (For example, the z-index is set to -1 on all tabs, instead of 0. As well, our mat-tab-nav-panel is buried deep in a <app-header> component, and providing it with a tabPanel that includes the router-outlet is... difficult.)

**Comment by tonybaroneee** [comment]
+1 to everything above. @EliezerB123 My workaround involves having an empty tabPanel:

```
<nav mat-tab-nav-bar [tabPanel]="tabPanel">
    <a mat-tab-link ...>Link 1</a>
    ...
</nav>
<mat-tab-nav-panel #tabPanel></mat-tab-nav-panel>
```

**Comment by avine** [comment]
Please fix this issue. The proposed workaround does not work in terms of accessibility as pointed by @EliezerB123 .
Guys, you already got the affected lines code as pointed by @epelc
